buildifier/utils/utils.go

```diff
@@ -115,13 +116,39 @@ func SplitFilePath(filename string) (workspaceRoot, pkg string) {
 	return workspaceRoot, pkg
 }
 
+// getFileReader returns a *FileReader object that reads files from the local
+// filesystem if the workspace root is known.
+func getFileReader(workspaceRoot string) *warn.FileReader {
+	if workspaceRoot == "" {
+		return nil
```

Comment:
is there anyway to return a 'no-op' Filereader that doesn't require any special nil handling?

Reply:
I chose to use a nil FileReader because 1. if a client doesn't need such functionality it doesn't need to provide a dummy FileReader object, just `nil`; 2. if a check doesn't make sense without a file reader it can easily check for `if fileReader == nil` and return.

Reply:
expecting/checking for nil is not a normal Go-idiom.

More often you provide a Noop struct/interface, and/or return (*foo, ok bool) where the bool tells you if the struct is valid.  This could be the case for an interface

Reply:
Should the callers that need to fix/lint a single file explicitly provide a dummy object that implements an interface with a single GetFile method? If so, should there be a singleton object, or should the callers explicitly create one, e.g. `warn.FixWarnings(..., warn.DummyFileReader{})`?

Reply:
Another question is even if I create an interface and several implementations (real, dummy), a caller may still pass `nil` as its value, and then   its methods will cause a segmentation fault error. I'll still need to check for nil to avoid that.

Buildifier is supposed to be used as a library, so by callers I actually mean other people who implement their own tools using buildifier. How can I enforce the usage of noop objects, especially given that the FileReader object is optional and using `nil` won't cause any breakages unless users actually try to run one of the multifile warnings?